pkg/controller.v1alpha3/experiment/experiment_controller.go

```diff
@@ -115,19 +115,26 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 // addWatch adds a new Controller to mgr with r as the reconcile.Reconciler
 func addWatch(mgr manager.Manager, c controller.Controller) error {
+	experimentType := experimentsv1alpha3.Experiment{}
+	experimentType.APIVersion = consts.APIVersionToWatch
+	trialType := trialsv1alpha3.Trial{}
+	trialType.APIVersion = consts.APIVersionToWatch
+	suggestionType := suggestionsv1alpha3.Suggestion{}
+	suggestionType.APIVersion = consts.APIVersionToWatch
```

Comment:
Why do we need it? I think watch can infer the APIVersion from the type.

Reply:
@gaocegege After testing with controllers of both versions, I found v1beta1 experiments can also be watched by v1alpha3 controller. Vice versa.

Reply:
Did you register two API versions?

Reply:
Which one is served/storaged? Ref https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition

Reply:
@gaocegege  I found my problem, thanks for documentation! It's really helpful.